Add lint

rubenarakelyan opened the pull request:
Add linting to manuals-frontend, and fix issues that it raises.

fofr commented:
Can we enable Sass linting too? (It comes with govuk-lint)

h-lame commented on Gemfile.lock:
```diff
@@ -257,3 +276,6 @@ DEPENDENCIES
   webmock (~> 2.1.0)
+
+BUNDLED WITH
+   1.12.5
```
Ruby 2.3.1 which this project uses is pinned to bundler `1.11.2` (see https://github.com/alphagov/govuk-puppet/blob/9a17813a00e18ad6c22db48ceda24d233d404b24/modules/govuk_rbenv/manifests/all.pp#L56) so this line shouldn't read `1.12.5`.

rubenarakelyan replied:
Fixed, didn't know about this!

h-lame commented:
:clap: for intent! It doesn't seem to pass on CI though, more linting errors to fix?  I'm not sure if we're still doing this, but in other projects I think we were setting up lint to run slightly differently on CI too - in `--diff` mode (and with some extra --format options for output) so we're only checking for lint problems in changed files (useful for when we bump linter and get new rules, or we add linter and don't want to add a big "fix all the things" noisy commit).

rubenarakelyan commented:
@fofr Yup, added sass linting
@h-lame CI showed some more linting errors that I fixed. I have changed the rake talk to not run by default, and the Jenkinsfile now runs both linters with the relevant flags set, whereas the rake task runs across the whole codebase.

I have made a load of sass fixes to the stylesheets. I have checked these visually and they look fine but a second eye over it would be great.

fofr commented on app/assets/stylesheets/_reset.scss:
```diff
@@ -1,6 +1,8 @@
-// flatten all browser defaults and styles inherited from gov.uk header
-
-/* Borrowed from http://meyerweb.com/eric/tools/css/reset/ */
+// Flatten all browser defaults and styles inherited from gov.uk header
```
It's weird we have a custom reset in here.

fofr commented on app/assets/stylesheets/application.scss:
```diff
@@ -8,8 +8,12 @@
 @import "reset";
 
+// some colours used below
+$manual-search-button-colour: #0b0c0c;
```
I think this can be set to `$black`:
https://github.com/alphagov/govuk_frontend_toolkit/blob/89168934d1455754f455c398ba5c350a484d9a2a/stylesheets/colours/_palette.scss#L50

(I like that the button variable has been extracted out as a concept)

fofr commented:
Really happy to see the Sass linting fixes :)
